README.md

```diff
@@ -103,4 +103,4 @@ used and maintained by [Actyx](https://www.actyx.com).
 - [Subspace](https://github.com/subspace/subspace) - Subspace Network reference implementation
 - [Substrate](https://github.com/paritytech/substrate) - Framework for blockchain innovation,
 used by [Polkadot](https://www.parity.io/technologies/polkadot/).
-- [Ursa](https://github.com/fleek-network/ursa) - Decentralized Content Delivery & Edge Compute.
+- [Taple](https://github.com/opencanarias/taple-core) - Sustainable DLT for asset and process traceability by [OpenCanarias](https://www.opencanarias.com/en/).
```

Comment:
Something seems to have gone wrong during the rebase / merge because this diff is showing up here as if it were in this PR?

Reply:
yeah...I think I pulled the upstream master into my branch at some point. let me see if I can clean this up.

Reply:
I tried to do an interactive rebase and drop the unwanted commits and they don't show up in the list of available commits to drop. So instead I created a new branch and used cherry-pick to recreate this PR's branch with just my commits and your review commits. The new branch is here: https://github.com/dhuseby/rust-libp2p/tree/add-kad-modechanged-event

I was think the best course of action at this point is to close this PR and file a new PR with the new clean branch. What do you think?

Reply:
> I was think the best course of action at this point is to close this PR and file a new PR with the new clean branch. What do you think?

Yes, sounds good!